### ISSUE TITLE
Focus is not called on draggable focus anymore.

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#693]
While working on https://github.com/phetsims/friction/issues/211, I noticed that the focus listeners on the GrabDragInteraction's draggable element were not firing. This means at least two listeners that we thought were firing on not:

https://github.com/phetsims/friction/blob/ffe03649b6e295da75095d889b25d591c9ced367/js/friction/view/book/BookNode.js#L112-L120

and also 

https://github.com/phetsims/scenery-phet/blob/7a6b0480e7b81dc39ec48dc369bd951f59df9523/js/accessibility/GrabDragInteraction.js#L430-L434

I'm not sure when this happened, but I feel like I trust myself enough to think that the friction alert was at some point firing (like back in 2018 when we were working on it).  That said. I can reproduce this bug with https://phet-dev.colorado.edu/html/friction/1.5.11/phet/friction_a11y_view.html on chrome and firefox.

I'll investigate.

**Comment by jessegreenberg** [comment]
It looks like the focus listeners are specifically blocked because of this in Input.js

```js
        // ignore any focusout callbacks if they are initiated due to implementation details in PDOM manipulation
        if ( this.display.blockFocusCallbacks ) {
          return;
        }
```

**Comment by zepumph** [comment]
That feels ok to me. Thinking through the actual definition of the inputListenerss, we shouldn't expect focus to be called after mutating a Node, where all the while that Node is focused. That doesn't really make sense.

I found that this patch fixes the problem.

```diff
Index: js/accessibility/GrabDragInteraction.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/accessibility/GrabDragInteraction.js b/js/accessibility/GrabDragInteraction.js
--- a/js/accessibility/GrabDragInteraction.js	(revision 5769b702fa042923742ce0826b0843c2bd7f5a7e)
+++ b/js/accessibility/GrabDragInteraction.js	(date 1628712791551)
@@ -356,10 +356,15 @@
         // if the draggable was just released, don't pick it up again until the next click event so we don't "loop"
         // and pick it up immediately again.
         if ( !guardKeyPressFromDraggable ) {
+
+          // blur as a grabbable
+          this.node.blur();
+
           this.turnToDraggable();
 
           this.numberOfKeyboardGrabs++;
 
+          // blur as a draggable
           this.node.focus();
 
           this.onGrab( event );

```

**Comment by zepumph** [comment]
You can test this out by running friction with `?sceneryLog=Input` and searching in the console for "focus"

**Comment by jessegreenberg** [comment]
That fix makes sense because I also noticed that the `listenersToAdd` are added after the Node mutates. So even without the block in Input.js I wouldn't expect the `focus` event to fire because the listener is added after focus changes from the `mutate`.

https://github.com/phetsims/scenery-phet/blob/7a6b0480e7b81dc39ec48dc369bd951f59df9523/js/accessibility/GrabDragInteraction.js#L594-L602

> The only reason we wouldn't want to do it this way is if there was a blur listener as part of listenersToGrabState that wanted to know when blur occurred moving to another Node, and not when activating the grab button.

Thats a good point.  But I can't think of anything better. Should we assert that listenersForGrabState does not include a `blur` listener?

**Comment by jessegreenberg** [comment]
I committed the patch, with the suggested assertion. @zepumph can you please review?

**Comment by zepumph** [comment]
@jessegreenberg and I went over the above commit. We agreed it made the most sense to allow blur in grabbable, but to note that. In the future we could pretty easily add a flag like `if (this.turningToDraggable)` that grabbable blur listeners can check on if they don't want to fire in this case. Closing

**Comment by samreid** [comment]
As discovered in https://github.com/phetsims/chipper/issues/946, there are TODOs in the code referring to this issue.  Reopening.

**Comment by jessegreenberg** [comment]
This TODO was added to identify the core problem of this issue. It should be removable now that this is fixed. I verified that the `focus` listener in the GrabDragInteraction is called so the TODO no longer applies and can definitely be removed now.
